recirq/benchmarks/rep_rate/rep_rate_calculator.py

```diff
@@ -0,0 +1,263 @@
+"""
+  Class to calculate latency and repetition rate.
+
+  This class should replicate the acquisition of date for
+  rep rate and latency for processor data sheets.
+"""
+import random
+import cirq
+import cirq.google as cg
+import time
+import sympy
+
+from typing import List, Optional, Tuple
+
+
+class RepRateCalculator:
+    """Class to perform simple timing experiments using hardware.
+
+    This class calculates the repetition rate for various
+    types of circuits.  This is the approximate type per repetition
+    (or shot) to execute this on the device.  This rate varies widely
+    based on width, depth, and amount of sweeps/batching in the
+    circuit.  This class will help test various rates on the device.
+
+    This class is used to generate statistics on the processor datasheets
+    for latency and repetition rates.
+    """
+
+    def __init__(self,
+                 project_id: str,
+                 processor_id: str,
+                 gate_set: Optional[cg.SerializableGateSet] = None,
+                 device: Optional[cirq.Device] = None,
```

Comment:
Consider having the default constructor take explicit, required `(device, sampler, qubits, gate)` with a factory function (`classmethod`) which takes a processor_id and constructs using engine. I see `self.engine` is not saved as an attribute to this class.

Reply:
Done.